sdk/core/core-rest-pipeline/test/node/bearerTokenAuthenticationPolicyChallenge.spec.ts

```diff
@@ -386,4 +387,80 @@ describe("bearerTokenAuthenticationPolicy with challenge", function() {
     assert.ok(error);
     assert.equal(error?.message, "Failed sendRequest error");
   });
+
+  it("tests that the scope and the claim have been passed through to getToken correctly - with @azure/core-client's authorizeRequestOnClaimChallenge", async function () {
+    const expected = {
+      scope: ["http://localhost/.default"],
+      challengeClaims: JSON.stringify({
+        access_token: { foo: "bar" }
+      })
+    };
+
+    const pipelineRequest = createPipelineRequest({ url: "https://example.com" });
+    const responses: PipelineResponse[] = [
+      {
+        headers: createHttpHeaders({
+          "WWW-Authenticate": `Bearer scope="${expected.scope[0]}", claims="${encodeString(
+            expected.challengeClaims
+          )}"`
+        }),
+        request: pipelineRequest,
+        status: 401
+      },
+      {
+        headers: createHttpHeaders(),
+        request: pipelineRequest,
+        status: 200
+      }
+    ];
+
+    const expiresOn = Date.now() + 5000;
+    const getTokenResponse = { token: "mock-token", expiresOnTimestamp: expiresOn };
+    const credential = new MockRefreshAzureCredential([getTokenResponse]);
+
+    const pipeline = createEmptyPipeline();
+    let firstRequest: boolean = true;
+    const bearerPolicy = bearerTokenAuthenticationPolicy({
+      // Intentionally left empty, as it should be replaced by the challenge.
+      scopes: [],
+      credential,
+      challengeCallbacks: {
+        async authorizeRequest({ request, getAccessToken }) {
+          if (firstRequest) {
+            firstRequest = false;
+            // send first request without the Authorization header
+          } else {
+            const token = await getAccessToken([], {});
+            request.headers.set("Authorization", `Bearer ${token}`);
+          }
+        },
```

Comment:
I’ll remove this callback in a bit. This shouldn’t be needed to test CAE challenges.